codex/sales/reservations.nim

```diff
@@ -221,20 +239,19 @@ proc updateImpl(
 
   return success()
 
-proc update*(
-  self: Reservations,
-  obj: Reservation): Future[?!void] {.async.} =
-  return await self.updateImpl(obj)
-
-proc update*(
+proc updateAvailability(
```

Comment:
I needed a non-locking way to update availability for functions that already have acquired a lock in order to avoid deadlocks. I thought about adding a flag "to lock or not" to the `update()` function, but since it is a public function, I suppose it should always lock externally. If you have any other suggestions on how to make this cleaner, please feel free to let me know.

Reply:
When is a lock not needed?

Reply:
Everywhere where `updateAvailability` is used 🤪 
But for example in `deleteReservation` and `createReservation`... These have already the lock acquired...